File: country_spa - front/src/app/model/comment.ts
export interface Comment {
    id: string;
    commentText: string;
    createdAt: Date;
    cityName: string;
    userName: string;
}

export interface CreateCommentResponse {
    id: string;
    createdAt: Date;
    commentText: string;
}


export interface CreateComment {
    commentText: string;
    userMail: string;
    cityId: string;
}

File: country_spa - back/src/main/java/org/proway/student/country_spaback/domain/Comment.java
package org.proway.student.country_spaback.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Table(name = "comment")
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnore
    private User user;

    @ManyToOne
    @JoinColumn(name = "city_id")
    @JsonIgnore
    private City city;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "comment_text")
    private String commentText;

}
